caches/clan.go

```diff
@@ -0,0 +1,91 @@
+package caches
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+	"github.com/topfreegames/khan/models"
+)
+
+// ClansSummariesCache represents a cache for the RetrieveClansSummaries operation
+type ClansSummariesCache struct {
+	Cache                 *gocache.Cache
+	TimeToLive            time.Duration
+	TimeToLiveRandomError time.Duration
+}
+
+// GetClansSummaries returns a summary of the clans details for a given list of clans by their game
+// id and public ids
+func (c *ClansSummariesCache) GetClansSummaries(db models.DB, gameID string, publicIDs []string) ([]map[string]interface{}, error) {
+	resultMap := c.getCachedClansSummaries(gameID, publicIDs)
+	err := c.getAndCacheClansSummaries(db, gameID, resultMap)
+	if err != nil {
+		if _, ok := err.(*models.CouldNotFindAllClansError); !ok {
+			return nil, err
+		}
+	}
+	var result []map[string]interface{}
+	for _, publicID := range publicIDs {
+		if resultMap[publicID] != nil {
+			result = append(result, resultMap[publicID])
+		}
+	}
+	return result, err
+}
+
+func (c *ClansSummariesCache) getClanSummaryCacheKey(gameID, publicID string) string {
+	return fmt.Sprintf("%s/%s", gameID, publicID)
+}
+
+func (c *ClansSummariesCache) getClanSummaryCache(gameID, publicID string) map[string]interface{} {
```

Comment:
The purpose of `getClanSummaryCache` and `setClanSummaryCache` is to isolate the `c.Cache` usage from the rest of the `caches.ClansSummaries` code. If we decide to use a different cache object in the future (like a connection to Redis, for example) it will be easier to refactor.

Reply:
Up to you, but I would inline it and worry about refactoring only if/when we needed it. (My guess is that calling Redis inline would be fine too.)